specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json

```diff
@@ -1095,34 +1107,115 @@
       },
       "description": "Class representing a Traffic Manager HeatMap traffic flow properties."
     },
-    "HeatMapProperties": {
+    "HeatMapPropertyBase": {
+      "type": "object",
+      "discriminator": "heatMapType",
```

Comment:
That's not exactly the way to do it, but we're close :)
- HeatMapModel should declare id, name, and type and nothing more
- HeatMapModel should have the discriminator on "type", since this is the actual value that discriminates (`heatMapType` does not exist in the JSON).
- For each type, you need two classes (take Country as example)
  - HeatMapModelCountry that is a allOf of HeatMapModel and has the x-discriminator-value, and contains one node properties described in a class HeatMapModelCountryProperties. This last one does not talk about discriminator

Hope this clarify a bit, it's easier to see now we made progress and have examples.